### PR TITLE
Constraint revise

### PR DESCRIPTION
**Pull request by ocots**
Il faut finaliser ça.

### CODE REVIEW

**Comment by ocots** [comment]
There is a problem with the documentation. Maybe because of the basic example in functional form.

**Comment by ocots** [comment]
I think this does not work in this version:

```julia
constraint!(ocp, :initial, [ -1, 0 ])           # initial condition
constraint!(ocp, :final,   [  0, 0 ])           # final condition
```

since there is no more

```julia
constraint!(
    ocp::OptimalControlModel,
    type::Symbol,
    val::Union{Real, AbstractVector{<:Real}}
)
constraint!(
    ocp::OptimalControlModel,
    type::Symbol,
    val::Union{Real, AbstractVector{<:Real}},
    label::Symbol
)
```

**Comment by ocots** [comment]
Do we want:

```julia
function constraint!(ocp::OptimalControlModel, type::Symbol, val::ctVector, label::Symbol=__constraint_label())
    __constraint!(ocp, type, rg=nothing, f=nothing, lb=val, ub=val, label=label)
    nothing # to force to return nothing
end
```

or since we have:

```julia
function constraint!(ocp::OptimalControlModel, 
    type::Symbol, 
    lb::Union{ctVector,Nothing}, 
    ub::Union{ctVector,Nothing}, 
    label::Symbol=__constraint_label())

    __constraint!(ocp, type, rg=nothing, f=nothing, lb=lb, ub=ub, label=label)
    nothing # to force to return nothing

end
```

In the first case, we can write

```julia
constraint!(ocp, :initial, [ 1, 2, 1 ])
```

while if we do not have the first function then we should write


```julia
constraint!(ocp, :initial, [ 1, 2, 1 ], [ 1, 2, 1 ])
```

**Comment by jbcaillau** [comment]
@ocots @BaptisteCbl right, we should check and close it. plus [the issue](https://github.com/control-toolbox/CTBase.jl/issues/115) for `time!`.

**Comment by ocots** [comment]
I keep only the method `constraint!` with keywords.

**Comment by jbcaillau** [comment]
> I keep only the method `constraint!` with keywords.

@ocots OK 👍🏽 Note that there are indeed consistency checks in the *general* `(__)constraint` function (check the last case of the pattern matching below):
https://github.com/control-toolbox/CTBase.jl/blob/a7e20237fa1a74990879ba067ca8b153f95138fb/src/model.jl#L676

To me, there should only be some old functional syntax tests to update (some of them are already preventing the doc CI). 

Then we must also take care of `time!` along the same line.

**Comment by ocots** [comment]
Up to now there is only one `constraint!` method. 

Note that for the unitary tests, I have defined `__constraint!` to keep previous tests. See [this file](https://github.com/control-toolbox/CTBase.jl/blob/constraint_revise/test/runtests.jl).

**Comment by ocots** [comment]
I think that constraint are revised (if the CI is ok) and we can make another branch for `time!` revision.

File: docs/src/api-print.md
Comment (jbcaillau) [inline docs/src/api-print.md:1]: @ocots is it really the only place where this is used?

File: src/model.jl
Comment (jbcaillau) [inline src/model.jl:1]: @ocots I see some blue style here, well done 👍🏽 SNCF time 🙂

Comment (jbcaillau) [inline src/model.jl:1]: @ocots very nice tests update 👍🏽

File: test/test_model.jl
Comment (jbcaillau) [inline test/test_model.jl:1]: @ocots well done 👍🏽👍🏽 rapid test update providing `__constraint!` that preserves the old functional style and is mapped (and so tests) the new `constraint!` with named args.

**Pull request merged**